config/routes/api.js

```diff
@@ -1,11 +1,40 @@
 var express             = require('express')
   , errorHandler        = require('express-error-middleware')
+  , multer              = require('multer')
+  , path                = require('path')
+  , crypto              = require('crypto')
   , helpers             = require('./helpers')
   , ensureAuthenticated = helpers.ensureAuthenticated
   , articles            = require('../../app/controllers/articles')
   , unions              = require('../../app/controllers/unions');
 
+var EXTENSION_LOOKUP = {
+  'image/jpeg': '.jpg',
+  'image/png': '.png',
+  'image/gif': '.gif'
+};
+
 var router = express.Router();
+var storage = multer.diskStorage({
+  destination: function(req, file, cb) {
+    cb(null, path.resolve('public', 'images', 'unions'));
+  },
+  filename: function (req, file, cb) {
+    // multer doesn't figure out the extension, so let's make a guess:
+    var extension = EXTENSION_LOOKUP[file.mimetype];
+    var filename = crypto.randomBytes(16).toString('hex');
```

Comment:
this is pretty much the same behavior as multer does by default (the random stuff)